cluster/gce/configure-vm.sh

```diff
@@ -559,11 +559,8 @@ if [[ -z "${is_push}" ]]; then
   set-kube-env
   [[ "${KUBERNETES_MASTER}" == "true" ]] && mount-master-pd
   create-salt-pillar
-  if [[ "${KUBERNETES_MASTER}" == "true" ]]; then
-    create-salt-master-auth
```

Comment:
I think I can see in the salt why this is happening. @zmerlynn 

configure-vm.sh sets the apiserver grain on the minions but not on the master (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/configure-vm.sh#L541). So when salt is determining what address to pass the kubelet for the master, it hits the third or fourth case in `cluster/saltbase/salt/kubelet/default` which ends up picking the 10. address (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/saltbase/salt/kubelet/default#L11-L12 or https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/saltbase/salt/kubelet/default#L14-L15). I can't tell which case is happening, but it seems likely that one (or both) of them would result in the internal address being used.

That means that this change won't help because it isn't setting the apiserver grain on the master VM. 

In addition, in #6949 I'm explicitly detaching the kubelet on the master from the cluster, so this problem will go away. When we re-add the option to attach it, we should plumb the appropriate configuration bits so that the master looks just like any other node in the cluster.